src/net.cpp

```diff
@@ -2331,6 +2332,11 @@ bool CAddrDB::Read(CAddrMan& addr)
 
     CDataStream ssPeers(vchData, SER_DISK, CLIENT_VERSION);
 
+    return Read(addr, ssPeers, hashIn);
+}
+
+bool CAddrDB::Read(CAddrMan& addr, CDataStream& ssPeers, uint256& hashIn)
+{
     // verify stored checksum matches input data
     uint256 hashTmp = Hash(ssPeers.begin(), ssPeers.end());
```

Comment:
> YMMV, but in my experience tmp files in unittests cause hard to debug test failures due to filesystem weirdness and added state (file deletes sometimes fail).

I agree. It's somewhat preferable to not clutter files around the place.
On the other hand it actually tests the filesystem interaction then :)

> I share your concern as this would require a big diff when all the instances of File are replaced with mockable File.

C++ has some support for this by replacing inputs streams by string streams, unfortunately - or not, I'm not entirely up to date of the pros and cons of it - we don't really use C++'s file API.

We mainly use `fopen` and `FILE*`. Note that on some platforms (including Linux and MacOSX IIRC) for FILE* there is: [fmemopen, open_memstream](http://linux.die.net/man/3/open_memstream). May be interesting.